cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -247,12 +247,14 @@ func SetDefaults_KubeletConfiguration(obj *MasterConfiguration) {
 	obj.KubeletConfiguration.BaseConfig.Authorization.Mode = kubeletconfigv1beta1.KubeletAuthorizationModeWebhook
 
 	// Let clients using other authentication methods like ServiceAccount tokens also access the kubelet API
-	// TODO: Enable in a future PR
-	// obj.KubeletConfiguration.BaseConfig.Authentication.Webhook.Enabled = utilpointer.BoolPtr(true)
+	obj.KubeletConfiguration.BaseConfig.Authentication.Webhook.Enabled = utilpointer.BoolPtr(true)
 
 	// Disable the readonly port of the kubelet, in order to not expose unnecessary information
-	// TODO: Enable in a future PR
-	// obj.KubeletConfiguration.BaseConfig.ReadOnlyPort = 0
+	obj.KubeletConfiguration.BaseConfig.ReadOnlyPort = 0
```

Comment:
Note that `ReadOnlyPort` is already `0` by default when you load a config from a file, or use dynamic config.
Similar for a few others too, like Webhook auth. The `v1beta1` defaulter is the canonical source of defaults, and we maintain the old defaults for flags via `applyLegacyDefaults` in `cmd/kubelet/app/options/options.go`.

So if you're just generating the default Kubelet config, then you'll get the file/dynamic config defaults if you do it like `NewKubeletConfiguration` in `options.go`.

Ah, noticed above you're doing this: 
```
if obj.KubeletConfiguration.BaseConfig == nil {
	obj.KubeletConfiguration.BaseConfig = &kubeletconfigv1beta1.KubeletConfiguration{}
 }
```

You might want to run it through the defaulter, like `NewKubeletConfiguration` in `options.go`.

Reply:
We do run these through the defaulter, see later down the func where we construct the scheme.
But I like to enforce this here as well (double-checking basically to be safe), and be explicit with what we require. Does that make sense? Yeah I saw these are way more secure by default when not using the flags :+1:

Reply:
Yup, that's totally reasonable.